### PR TITLE
Mentee edit form

### PR DESCRIPTION
**Pull request by MichaelS42**
trying to implement edit form into redux

### CODE REVIEW

**Comment by Brodt7258** [comment]
[get the mentee edit form hooked into redux/api](https://trello.com/c/izuPQ4Gd/30-get-the-mentee-edit-form-hooked-into-redux-api)

File: src/redux/StudentEdit.redux/StudentEdit.action.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import Axios from 'axios';
+
+export const EDIT_USER_START = 'EDIT_USER_START';
```
Comment (Brodt7258) [inline src/redux/StudentEdit.redux/StudentEdit.action.js:3]: let's move these over to a file named `StudentEdit.types.js`

Comment (Brodt7258) [reply src/redux/StudentEdit.redux/StudentEdit.action.js:3]: also, this file should named `StudentEdit.actions.js`, plural

File: src/redux/StudentEdit.redux/StudentEdit.reducer.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import {
+  EDIT_USER_START,
+  EDIT_USER_SUCCESS,
+  EDIT_USER_FAILURE,
+} from './StudentEdit.action';
+
+const userEditInitialState = {
+  editingUser: [],
```
Comment (Brodt7258) [inline src/redux/StudentEdit.redux/StudentEdit.reducer.js:8]: I don't think we need this variable. Unless I'm mistaken about what it's for, we're going to keep the form state local to StudentInfo.js

File: src/redux/StudentEdit.redux/StudentEdit.action.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+export const EDIT_USER_SUCCESS = 'EDIT_USER_SUCCESS';
+export const EDIT_USER_FAILURE = 'EDIT_USER_FAILURE';
+
+export const editUser = (id, { user_id, name }) => (dispatch) => {
```
Comment (Brodt7258) [inline src/redux/StudentEdit.redux/StudentEdit.action.js:7]: I'd actually skip the axios stuff for the moment, have it just edit the local mock data instead. the overall structure of this action looks good though. let me know if you want a hand doing the mock-edits.

File: src/redux/StudentEdit.redux/StudentEdit.action.js (hunk 1 above)
Comment (MichaelS42) [reply src/redux/StudentEdit.redux/StudentEdit.action.js]: done

File: src/redux/StudentEdit.redux/StudentEdit.reducer.js (hunk 2 above)
Comment (MichaelS42) [reply src/redux/StudentEdit.redux/StudentEdit.reducer.js]: removed variable

File: src/redux/StudentEdit.redux/StudentEdit.action.js (hunk 3 above)
Comment (MichaelS42) [reply src/redux/StudentEdit.redux/StudentEdit.action.js]: took out axios

**Review by LittleBoGeek — APPROVED** [review]
Nice job. Everything looks good.

**Review by PaulMEdwards — APPROVED** [review]
Form editing works and looks nice with the rest of the design now.

I did notice that the Student Info Display & Search Bar results don't update immediately after submitting the edits. This can be addressed later by hooking into the Redux store.

**Pull request merged**